client/state/login/actions.js

```diff
@@ -244,7 +244,19 @@ export const loginSocialUser = ( socialInfo, redirectTo ) => dispatch => {
 			dispatch( {
 				type: SOCIAL_LOGIN_REQUEST_SUCCESS,
 				redirectTo: get( response, 'body.data.redirect_to' ),
+				data: get( response, 'body.data' )
 			} );
+
+			if ( get( response, 'body.data.two_step_notification_sent' ) === 'sms' ) {
```

Comment:
Non-SMS 2FA don't need to fire `TWO_FACTOR_AUTHENTICATION_SEND_SMS_CODE_REQUEST_SUCCESS`, because it's just for showing the "SMS successfully sent" message.